pallets/pallet-lottery/src/lib.rs

```diff
@@ -0,0 +1,1077 @@
+                        Some(our_request) if matches!(our_request.action, pallet_parachain_staking::DelegationAction::Revoke(_)) => {
+                            if T::BlockNumber::from(our_request.when_executable) > now {
+                                log::error!("Collator {:?} finished our unstaking timelock but not the pallet_parachain_staking one. leaving in queue", collator.account.clone());
+                                return true;
+                            };
+                            our_request.action.amount()
+                        }
+                        _ => {
+                                log::error!( "Expected revoke_delegation request not found on collator {:?}. Leaving in withdraw queue", collator.account.clone() );
+                                return true;
+                            }
+                    };
+                    // Ensure the pallet has enough gas to pay for this
+                    let fee_estimate : BalanceOf<T> = T::EstimateCallFee::estimate_call_fee(&pallet_parachain_staking::Call::execute_delegation_request { delegator: Self::account_id() , candidate: collator.account.clone()  }, None::<u64>.into());
+                    if Self::surplus_funds() <= fee_estimate{
+                        log::warn!("could not finish unstaking delegation because the pallet is out of funds to pay TX fees. Skipping");
+                        return true;
+                    };
+                    match pallet_parachain_staking::Pallet::<T>::execute_delegation_request(
+                        RawOrigin::Signed(Self::account_id()).into(),
+                        Self::account_id(),
+                        collator.account.clone(),
+                    ){
+                        Err(e) => {
+                            log::error!("Collator finished unstaking timelock but could not be removed with error {:?}",e);
+                            true
+                        },
+                        Ok(_) => {
+                            // collator was unstaked
+                            SumOfDeposits::<T>::mutate(|balance| *balance = (*balance).saturating_sub(balance_to_unstake));
+                            log::debug!("Unstaked {:?} from collator {:?}",balance_to_unstake,collator.account.clone());
+                            // don't retain this collator in the unstaking collators vec
+                            false
+                        },
+                    }
+            });
+            if original_len != unstaking.len() {
+                log::debug!(
+                    "Finished unstaking {:?} out of {:?} requests",
+                    original_len - unstaking.len(),
+                    original_len
+                );
+                UnstakingCollators::<T>::put(unstaking);
+            }
+        }
+
+        #[named]
```

Comment:
Interesting, I never see such macro before.

Reply:
it enables `log::trace!(function_name!());`